docs/quality_assurance.md

```diff
@@ -0,0 +1,169 @@
+
+For an overview of the Interchain Security workflow, have a look at [the diagrams](#interchain-security-workflow) at the end of this document.
+
+## Quality Assurance Concerns
+
+### Software Engineering
+
+| ID | Concern | Code Review | Automatic Tools | Unit Testing |
+| -- | ------- | ----------- | --------------- | ------------ |
+| 1.01 | Unexpected panics | `Scheduled` | `??` | `??` |
+| 1.02 | Handling errors | `Scheduled` | `gosec` | `??` |
+| 1.03 | Accessing store (setters, getters, iterators) | `Scheduled` | `??` | `Partial coverage` |
+| 1.04 | Serialization / deserialization | `Scheduled` | `??` | `??` |
+| 1.05 | Storage leaks | `Scheduled` | `NA` | `??` |
+
+### Integration with IBC
+
+Interchain Security is an IBC application and, thus, it relies on IBC to establish a separate channel between the provider chain and every consumer chain. Interchain Security relies on the IBC v3.0 golang [implementation](https://github.com/cosmos/ibc-go/tree/v3.0.0).
+
+IBC packets:
+- ValidatorSetChangePacket
+- MaturedVSCPacket
+- SlashPacketData
+
+| ID | Concern | Code Review | Unit Testing | Diff. testing | Testnet |
+| -- | ------- | ----------- | ------------ | ------------- | ------- |
+| 2.01 | Create IBC clients | `Scheduled` (ibc-go team) | `Done` | `Future work` | `Scheduled` | 
+| 2.02 | Getting consumer `UnbondingPeriod` from IBC client | `Scheduled` (ibc-go team) | `??` | `NA` | `NA` | 
+| 2.03 | Create CCV channel (handshake) | `Scheduled` (ibc-go team) | `Done` | `Future work` | `Scheduled` | 
+| 2.04 | Sending IBC packets <br /> - see `x/ccv/utils/utils.go:SendIBCPacket()` | `Scheduled` (ibc-go team) | `Done` | `Done` | `Scheduled` |
+| 2.05 | Handling acknowledgments | `Scheduled` (ibc-go team) | `Partial coverage` | `Scheduled` | `Scheduled` |
+| 2.06 | Handling timeouts | `Scheduled` (ibc-go team) | `Partial coverage` | `Future work` | `Scheduled` |
+| 2.07 | **Handling IBC client expiration** | `Scheduled` (ibc-go team) <br /> high priority | `??` | `Future work` | `Scheduled` |
+| 2.08 | ICS-20 channel creation | `Scheduled` (ibc-go team) | `??` | `Future work` | `Scheduled` | 
+| 2.09 | ICS-20 transfer | `Scheduled` (ibc-go team) | `??` | `NA` | `Scheduled` | 
+| 2.10 | Changes in IBC-GO testing suite | `Scheduled` (ibc-go team) | `NA` | `Partial coverage` | `NA` | 
+
+### Integration with Cosmos SDK
+
+A prerequisite of the code review is to open a PR with all the [SDK changes](https://github.com/cosmos/cosmos-sdk/tree/interchain-security-rebase) needed by Interchain Security.
+
+| ID | Concern | Code Review | Unit Testing | Diff. testing | Testnet | 
+| -- | ------- | ----------- | ------------ | ------------- | ------- | 
+| 3.01 | Changes to staking module | `Scheduled` (sdk team) | `Partial coverage` <br /> see [unbonding_test.go](../x/ccv/provider/unbonding_test.go) <br />  redelegation and validator unbonding missing | `Partial coverage` | `Scheduled` | 
```

Comment:
There are also tests in the Cosmos SDK repo, should they be included here?

Reply:
That would be a good idea.